inference-engine/tests/functional/plugin/shared/include/behavior/test_plugin.hpp

```diff
@@ -83,20 +59,15 @@ TEST_P(BehaviorTests, canNotLoadNetworkWithoutWeights) {
 TEST_P(BehaviorTests, pluginDoesNotChangeOriginalNetwork) {
     // Skip test according to plugin specific disabledTestPatterns() (if any)
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
-    const std::string name_a = "a.xml";
-    const std::string name_b = "b.xml";
     auto param = GetParam();
 
     InferenceEngine::CNNNetwork cnnNet(function);
-    InferenceEngine::CNNNetwork execGraph;
-    cnnNet.serialize(name_a);
+    InferenceEngine::CNNNetwork cnnNetCloned(InferenceEngine::cloneNetwork(cnnNet));
 
     ASSERT_NO_THROW(ie->LoadNetwork(cnnNet, targetDevice, configuration));
-    cnnNet.serialize(name_b);
 
-    compare_two_files_lexicographically(name_a, name_b);
-    EXPECT_EQ(0, std::remove(name_a.c_str()));
-    EXPECT_EQ(0, std::remove(name_b.c_str()));
+    // compare 2 networks
+    compare_functions(cnnNetCloned.getFunction(), cnnNet.getFunction());
```

Comment:
It can be incorrect. for example if we change constant operations the test will be passed.

Reply:
what is the proposal?

Reply:
Deep copy of nGraph function or read/create nGraph function 2 times to guaranty that we have 2 equal function without common weights.

Reply:
Used second approach, thanks!